packages/browser-wallet/src/popup/pages/Account/TransactionLog/TransactionList.tsx

```diff
@@ -140,6 +143,44 @@ function InfiniteTransactionList({
     );
 }
 
+/**
+ * Update an existing list of descending transactions with a new list of ascending transactions,
+ * ensuring that any overlaps are updated with the values from the new list of transactions.
+ */
+function updateTransactionsWithNewTransactions(
+    existingTransactions: BrowserWalletTransaction[],
+    newTransactions: BrowserWalletTransaction[]
+) {
+    const existingHashes = existingTransactions.map((trs) => trs.transactionHash);
+    const transactionUpdates: Record<string, BrowserWalletTransaction> = {} as Record<string, BrowserWalletTransaction>;
+
+    // Collect updates to existing transactions.
+    newTransactions
```

Comment:
You could do a split instead of a filter here, so you can avoid going through the list again later. :thinking:

Reply:
I have used the `partition` function instead.